taxcalc/functions.py

```diff
@@ -469,7 +476,7 @@ def StdDed(DSI, _earned, STD, e04470, e00100, e60000,
     # Apply regular standard deduction if not dependent or compulsory itemizer
     if (DSI == 1):
         c04100 = min(STD[MARS - 1], c15100)
-    elif _compitem == 1 or (3 <= MARS and MARS <= 6 and MIDR == 1):
+    elif MIDR == 1:
```

Comment:
MIDR, Married Filing Separately Itemized Deductions Requirement Indicator, is on means that it would be necessary to itemize when filing separately, which implies the rest conditions.